cmd/mt-kafka-mdm-sniff/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"text/template"
+
+	inKafkaMdm "github.com/raintank/metrictank/input/kafkamdm"
+	"github.com/raintank/metrictank/stats"
+	"github.com/raintank/worldping-api/pkg/log"
+	"github.com/rakyll/globalconf"
+	"gopkg.in/raintank/schema.v1"
+)
+
+var (
+	confFile = flag.String("config", "/etc/raintank/metrictank.ini", "configuration file path")
+	format   = flag.String("format", "{{.Part}} {{.OrgId}} {{.Id}} {{.Name}} {{.Metric}} {{.Interval}} {{.Value}} {{.Time}} {{.Unit}} {{.Mtype}} {{.Tags}}", "template to render the data with")
+)
+
+type Data struct {
+	Part int32
+	schema.MetricData
+}
+
+type inputPrinter struct {
+	template.Template
+	data Data
+}
+
+func newInputPrinter(format string) inputPrinter {
+	tpl := template.Must(template.New("format").Parse(format + "\n"))
+	return inputPrinter{
+		*tpl,
+		Data{},
+	}
+}
+
+func (ip inputPrinter) Process(metric *schema.MetricData, partition int32) {
+	ip.data.MetricData = *metric
+	ip.data.Part = partition
+	err := ip.Execute(os.Stdout, ip.data)
+	if err != nil {
+		log.Error(0, "executing template: %s", err)
+	}
+}
+
+func main() {
+	flag.Usage = func() {
+		fmt.Fprintln(os.Stderr, "mt-kafka-mdm-sniff")
+		fmt.Fprintln(os.Stderr)
+		fmt.Fprintln(os.Stderr, "Inspects what's flowing through kafka (in mdm format) and reports it to you")
+		fmt.Fprintf(os.Stderr, "\nFlags:\n\n")
+		flag.PrintDefaults()
+	}
+	flag.Parse()
+	log.NewLogger(0, "console", fmt.Sprintf(`{"level": %d, "formatting":false}`, 2))
+	instance := "mt-kafka-mdm-sniff" + strconv.Itoa(rand.Int())
+
+	// Only try and parse the conf file if it exists
+	path := ""
+	if _, err := os.Stat(*confFile); err == nil {
```

Comment:
NewWithOptions can return various kinds of errors (file not found, parse error / invalid data, etc). All but the first are critical (user) errors, the first is not really an error in our use case since config file is optional.  by doing it this way we can cleanly separate what are real errors the user should fix and die on them.

note also that inspection of returned errors (by type, or string value) is considered an anti pattern in the go world. (see https://dave.cheney.net/2014/12/24/inspecting-errors for example)

also FWIW we apply this same mechanism elsewhere too:

```
 r os.Stat | rv vendor
1:metrictank.go:	if _, err := os.Stat(*confFile); err == nil {
2:cmd/mt-store-cat/main.go:	if _, err := os.Stat(*confFile); err == nil {
3:cmd/mt-kafka-mdm-sniff/main.go:	if _, err := os.Stat(*confFile); err == nil {
```